src/ansible_creator/subcommands/init.py

```diff
@@ -32,127 +43,116 @@ def __init__(
             config: App configuration object.
         """
         self._namespace: str = config.namespace
-        self._collection_name: str | None = config.collection_name
+        self._collection_name = config.collection_name or ""
         self._init_path: Path = Path(config.init_path)
         self._force = config.force
         self._creator_version = config.creator_version
         self._project = config.project
-        self._scm_org = config.scm_org
-        self._scm_project = config.scm_project
+        self._scm_org = config.scm_org or ""
+        self._scm_project = config.scm_project or ""
         self._templar = Templar()
         self.output: Output = config.output
 
-    def run(self: Init) -> None:  # noqa: C901
-        """Start scaffolding skeleton.
+    def run(self: Init) -> None:
+        """Start scaffolding skeleton."""
+        self._construct_init_path()
+        self.output.debug(msg=f"final collection path set to {self._init_path}")
+
+        if self._init_path.exists():
+            self.init_exists()
+        self._init_path.mkdir(parents=True, exist_ok=True)
+
+        if self._project == "collection":
+            self._collection()
+        elif self._project == "ansible-project":
+            self._playbook()
```

Comment:
Nitpick but I don't think these are good names for methods that do something

```suggestion
        if self._project == "collection":
            self.scaffold_collection()
        elif self._project == "ansible-project":
            self.scaffold_playbook()
```

Reply:
great suggestion :)

Reply:
I had originally done a little dispatcher pattern but since we haven't fully transitionaed away from the `ansible-project` name to `playbook` it didn't make any sense.  That said we can still eventually do getattr(f"_scaffold_{project}").  Not that big of a deal here with 2 but other subcommands will have more